nodejs/device/src/index.ts

```diff
@@ -175,11 +178,38 @@ class DeviceApp {
         );
       };
       this.twin.on("properties.desired.thief.pairing", listener);
-      const timer = setTimeout(
+      const pairingAttemptTimer = setTimeout(
         on_timeout,
         1000 * settings.pairingRequestSendIntervalInSeconds
       );
-    });
+
+      const finish = () => {
+        logger.warn("Pairing canceled.");
+        this.twin.removeListener("properties.desired.thief.pairing", listener);
+        clearTimeout(pairingAttemptTimer);
+        reject(new Canceled());
```

Comment:
suggestion: you do these `removeListener` `clearTimeout` operations in a few places.  do you want to add something like:
```
const cleanup = () => {
        this.twin.removeListener("properties.desired.thief.pairing", listener);
        clearTimeout(pairingAttemptTimer);
}
```
and call that everywhere you clean up? 

The benefit here and now is not all that great, but it makes maintenance much much easier, especially if every "cancellable promise" has a `cleanup` function that always gets called before resolving or rejecting.

Reply:
Good call.